src/react/hooks/useBackgroundQuery.ts

```diff
@@ -202,13 +206,16 @@ export function useBackgroundQuery<
     client.watchQuery(watchQueryOptions as WatchQueryOptions<any, any>)
   );
 
-  const [promiseCache, setPromiseCache] = React.useState(
-    () => new Map([[queryRef.key, queryRef.promise]])
-  );
-
+  const [wrapped, setWrappedQueryRef] = React.useState({
+    current: wrapQueryRef(queryRef, queryRef.promise),
+  });
+  if (unwrapQueryRef(wrapped.current)[0] !== queryRef) {
+    setWrappedQueryRef({ current: wrapQueryRef(queryRef, queryRef.promise) });
+  }
+  let wrappedQueryRef = wrapped.current;
```

Comment:
```suggestion
  const wrappedQueryRef = wrapped.current;
```

Sorry 😅.

Better yet, since you're not reading this value anywhere but the return from this hook, perhaps you can just use `wrapped.current` there?